todo/src/components/Todo.js

```diff
@@ -0,0 +1,114 @@
+import React, { Component } from 'react';
+import {Container, Row, Col, Form, FormGroup, Input, Button, Badge} from 'reactstrap';
+import {add , remove, toggle} from '../actions';
+import {connect} from 'react-redux';
+
+class Todo extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            newTodo: ''
+        };
+    }
+
+    handleInput = (e) => {
+        this.setState({ newTodo: e.target.value });
+    }
+    handleAdd = (e) => {
+        // console.log(e.type);
+        // console.log(e.key);
+        if(e.type === 'click' || e.key === 'Enter'){
+            e.preventDefault();
+            this.props.add(this.state.newTodo);
+            this.setState({ newTodo: '' });
+        }
+        e.stopPropagation();
+    }
+
+    handleToggle = (e) => {
+        const item = document.querySelector(`#${e.target.id}`);
+        this.props.toggle(item.dataset.index); // WORKS
+
+
+    }
+    
+    handleRemove = (e) => {
+        this.props.remove(e.target.dataset.index);
+        e.stopPropagation();
+    }
+
+    /** Make the delte button appear on hover */
+    handleMouseEnter = (e) => {
+        const index = e.target.dataset.index;
+        const badge = document.querySelector(`#badge${index}`)
+        badge.classList.remove("invisible");
+        badge.classList.add("visible");
+        
+    }
+    handleMouseLeave = (e) => {
+        const index = e.target.dataset.index;
+        const badge = document.querySelector(`#badge${index}`)
+        badge.classList.remove("visible");
+        badge.classList.add("invisible");
+
+    }
+
+    render() {
+        const todos = this.props.todos.slice().reverse();
+        let length = todos.length - 1;
+        return (
+            <Container>
+                <Form>
+                    <FormGroup>
+                        <Input
+                            onKeyPress={this.handleAdd} 
+                            onChange={this.handleInput}
+                            value={this.state.newTodo}
+                            name="new-task"
+                            placeholder="Add new task..."
+                        />
+                    </FormGroup>
+                    <Button onClick={this.handleAdd} >Add task to list</Button>
+                </Form>
+                <hr className="border"/>
+                {todos.map(
+                    (todo, i) => (<Row
+                                    style={{cursor: "pointer"}}
+                                    key={i}
+                                    onClick={this.handleToggle}
+                                    >
+                                        {/* data-finished="no" */}
+                                    <Col xs="12" sm={{ size: 10, offset:1 }}
+                                        className={todo.done ? 'text-left border border-info taskDone' : ('text-left border border-info')}
```

Comment:
Try `this.todo.done` and let me know if that fixes your problem with localStorage

Reply:
Hello Cole, I did not have the opportunity to check that yet (A lot of new stuff this week, lot to study and digest).
I'll try to come back to this later on (let you know).

Thank you